hibernate-core/src/main/java/org/hibernate/boot/internal/SessionFactoryOptionsBuilder.java

```diff
@@ -350,6 +352,7 @@ public SessionFactoryOptionsBuilder(StandardServiceRegistry serviceRegistry, Boo
 				CONVENTIONAL_JAVA_CONSTANTS, BOOLEAN, true );
 		this.procedureParameterNullPassingEnabled = cfgService.getSetting( PROCEDURE_NULL_PARAM_PASSING, BOOLEAN, false );
 		this.collectionJoinSubqueryRewriteEnabled = cfgService.getSetting( COLLECTION_JOIN_SUBQUERY, BOOLEAN, true );
+		this.omitJoinOfSuperclassTablesEnabled = cfgService.getSetting( OMIT_JOIN_OF_SUPERCLASS_TABLES, BOOLEAN, true );
```

Comment:
Am I the only one who wouldn't make it the default behavior but only a possible optimization knob for advanced users?

Reply:
How would we get broad testing if we wouldn't enable the optimization by default and allow it to be deactivated for further baking if necessary? I think 5.4 is a good candidate for enabling this by default.

Reply:
I'm not sure whether it can be allowed. The produced query results may be different for unconstrained inheritance hierarchies (a row that exists in a subclass table but not in a superclass table, which was previously omitted by the join but is now present in result sets). No blocker for me personally, and I can hardly believe that this will be a problem for anyone, but we seem to consider this very edge case at other locations so we may have to keep it in mind.